Tests/Integration/MobileCoinClientPublicApiIntTests.swift

```diff
@@ -1,573 +1,821 @@
+                    try await checkBalance()
+                    return
                 }
-            }
+            } catch {}
         }
+        try await checkBalance()
     }
 
-    func testCreateSignedContingentInput() throws {
-        try XCTSkipUnless(IntegrationTestFixtures.network.hasSignedContingentInputs)
+    func testSelfPaymentBalanceChangeFeeLevel() async throws {
+        let description = "Self payment"
+        try await testSupportedProtocols(description: description) {
+            try await self.selfPaymentBalanceChangeFeeLevel(transportProtocol: $0)
+        }
+    }
 
-        let description = "Creating signed contingent input"
-        try testSupportedProtocols(description: description) {
-            try createSignedContingentInput(transportProtocol: $0, expectation: $1)
+    func selfPaymentBalanceChangeFeeLevel(
+        transportProtocol: TransportProtocol
+    ) async throws {
+        let accountKey = try  IntegrationTestFixtures.createAccountKey(accountIndex: 1)
+        let client = try IntegrationTestFixtures.createMobileCoinClient(
+            accountKey: accountKey,
+            transportProtocol: transportProtocol)
+
+        func submitTransaction() async throws -> Balance {
+            let balance = try await client.updateBalances().mobBalance
+            print("Initial balance: \(balance)")
+            let transaction = try await client.prepareTransaction(
+                to: accountKey.publicAddress,
+                amount: Amount(100, in: .MOB),
+                fee: IntegrationTestFixtures.fee)
+            try await client.submitTransaction(transaction: transaction.transaction)
+            return balance
+        }
+
+        let initialBalance = try await submitTransaction()
+        var numChecksRemaining = 5
+        func checkBalance() async throws {
+            numChecksRemaining -= 1
+            print("Updating balance...")
+            let balance = try await client.updateBalances().mobBalance
+            print("Balance: \(balance)")
+
+            do {
+                let balancePicoMob = try XCTUnwrap(balance.amount())
+                let initialBalancePicoMob = try XCTUnwrap(initialBalance.amount())
+                guard balancePicoMob != initialBalancePicoMob else {
+                    guard numChecksRemaining > 0 else {
+                        XCTFail("Failed to receive a changed balance. initial balance: " +
+                                "\(initialBalancePicoMob), current balance: " +
+                                "\(balancePicoMob) picoMOB")
+                        return
+                    }
+
+                    try await Task.sleep(nanoseconds: UInt64(2 * 1_000_000_000))
+                    try await checkBalance()
+                    return
+                }
+            } catch {}
         }
+        try await checkBalance()
     }
 
-    func createSignedContingentInput(
```

Comment:
also this one ?

Reply:
Just added an async version of this back in.

Also, updated the logic for the sci tests to be enabled for testNet in addition to Alpha in Network Presets now that TestNet is on block version 3.